app/scripts/components/analysis/define/use-stac-collection-search.ts

```diff
@@ -156,14 +160,25 @@ function getInTemporalAndSpatialExtent(collectionData, aoi, timeRange) {
     const collection = collectionData.find(
       (c) => c.id === l.stacCol && stacApiEndpointUsed === c.stacApiEndpoint
     );
+
+    if(!collection.summaries || !!collection.summaries.length) {
+      // NOTE: Invalid data because collection does not include summaries
+      return l;
+    }
+
     return {
       ...l,
       isPeriodic: collection['dashboard:is_periodic'],
       timeDensity: collection['dashboard:time_density'],
       domain: collection.extent.temporal.interval[0],
-      timeseries: collection.summaries.datetime
+      timeseries: collection.summaries.datetime,
     };
   });
-
-  return filteredDatasetsWithCollections;
+  
+  const [collectionsWithSummaries, collectionsWithoutSummaries]: [DatasetWithCollections[], DatasetLayer[]] = filteredDatasetsWithCollections.reduce((result: [DatasetWithCollections[], DatasetLayer[]], d: DatasetWithCollections) => {
```

Comment:
I think we have some opportunities to make the code simple/coherent here.
So currently, the logic flows in a way 
1. get the collection data 2. filter the collection with too many requests in the hook. (near line90)

How do you think about following the current pattern? so `getInTemporalAndSpatialExtent` will return the collections, and we will filter out the collections' missing summaries in the hook, near line 90. 

Then the code can be 

```
const datasetWithTooManyRequests = ....
const datasetMissingSummaries = ....
const unselectableDatasetLayers = [...datasetWitlTooManyRequests, ...datasetMissingSummaries]
```
How do you think?